modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/history/upnext/UpNextHistoryDetailsPage.kt

```diff
@@ -0,0 +1,245 @@
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH40
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH70
+import au.com.shiftyjelly.pocketcasts.compose.loading.LoadingView
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.localization.helper.RelativeDateFormatter
+import au.com.shiftyjelly.pocketcasts.localization.helper.TimeHelper
+import au.com.shiftyjelly.pocketcasts.models.entity.BaseEpisode
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.settings.history.upnext.UpNextHistoryDetailsViewModel.UiState
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import java.text.DateFormat
+import java.util.Date
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun UpNextHistoryDetailsPage(
+    viewModel: UpNextHistoryDetailsViewModel = hiltViewModel(),
+    date: Long,
+    onRestoreClick: (restoreUpNext: () -> Unit) -> Unit,
+    onBackClick: () -> Unit,
+    bottomInset: Dp,
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    UpNextHistoryDetailsView(
+        date = date,
+        state = state,
+        onRestoreClick = { onRestoreClick { viewModel.restoreUpNext() } },
+        onBackClick = onBackClick,
+        bottomInset = bottomInset,
+    )
+}
+
+@Composable
+private fun UpNextHistoryDetailsView(
+    date: Long,
+    state: UiState,
+    onRestoreClick: () -> Unit,
+    onBackClick: () -> Unit,
+    bottomInset: Dp,
+) {
+    Column {
+        ThemedTopAppBar(
+            title = formatDate(date),
+            bottomShadow = true,
+            onNavigationClick = onBackClick,
+            actions = { iconColor ->
+                TextButton(
+                    onClick = onRestoreClick,
+                    enabled = state is UiState.Loaded,
+                ) {
+                    TextH50(text = stringResource(LR.string.restore))
+                }
+            },
+        )
+        when (state) {
+            is UiState.Loading -> LoadingView()
+            is UiState.Loaded -> {
+                UpNextHistoryEpisodes(
+                    state = state,
+                    bottomInset = bottomInset,
+                )
+            }
+
+            is UiState.Error -> UpNextHistoryErrorView()
+        }
+    }
+}
+
+@Composable
+private fun UpNextHistoryEpisodes(
+    state: UiState.Loaded,
+    bottomInset: Dp,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn(
+        modifier
+            .fillMaxHeight()
+            .padding(16.dp),
+        contentPadding = PaddingValues(bottom = bottomInset),
+    ) {
+        items(state.episodes) { episode ->
+            EpisodeRow(
+                episode = episode,
+                useEpisodeArtwork = state.useEpisodeArtwork,
+            )
+        }
+    }
```

Comment:
I can't see the bottom episode and the top episode is being cut off early.

![Screenshot_20250321_111102](https://github.com/user-attachments/assets/ce26a3e0-460c-4182-9630-78cc59af1822)

Reply:
> I can't see the bottom episode 

I see that bottom insets are added to the view. I can't reproduce it. Maybe padding updates for the top row fixed it. 

https://github.com/user-attachments/assets/8b97220e-6c6d-4dd6-9f37-579781528cdd